commands.js

```diff
@@ -511,6 +560,12 @@ var commands = exports.commands = {
 
 		if (!room.auth) room.auth = room.chatRoomData.auth = {};
 
+		if (room.isPersonal) {
+			if (!user.personalRooms) user.personalRooms = 0;
+			if (user.personalRooms + 1 >= MAX_PERSONAL_ROOMS) return this.sendReply("This user already has " + MAX_PERSONAL_ROOMS + " personal rooms.");
+			user.personalRooms++;
```

Comment:
The user object can be flushed really easily. It's okay for caches, but not for anything intended to be persistent. I'd say remove this limitation entirely.